test/e2e/plugin_sync/tmc/plugin_sync_tmc_lifecycle_test.go

```diff
@@ -560,28 +559,22 @@ var _ = f.CLICoreDescribe("[Tests:E2E][Feature:Plugin-Sync-TMC-lifecycle]", func
 		})
 
 		// Test case: g. TMC: list plugins and validate plugins info, make sure all plugins are installed as per mock response
-		It("list plugins and validate plugins being installed after context being created", func() {
+		It("Test case: g: list plugins and validate plugins being installed after context being created", func() {
 			installedPluginsListTMC, err = tf.PluginCmd.ListPluginsForGivenContext(contextNameTMC, true)
 			Expect(err).To(BeNil(), noErrorForPluginList)
 			Expect(len(installedPluginsListTMC)).Should(Equal(len(pluginsToGenerateMockResponse)), numberOfPluginsSameAsNoOfPluginsInfoMocked)
 			Expect(f.CheckAllPluginsExists(installedPluginsListTMC, pluginsToGenerateMockResponse)).Should(BeTrue(), pluginsInstalledAndMockedShouldBeSame)
 		})
 
 		// Test case: h. validate plugin list consistancy, it should sort by context name always
-		It("list plugins and validate plugins being installed after context being created", func() {
+		It("Test case: h: list plugins and validate plugins being installed after context being created", func() {
 			// check multiple times, the order should be consistent
 			for j := 0; j < 5; j++ {
 				installedPlugins, err := tf.PluginCmd.ListInstalledPlugins()
 				Expect(err).To(BeNil(), noErrorForPluginList)
 				Expect(totalInstalledPlugins).Should(Equal(len(installedPlugins)), "total installed plugins count should be equal to plugins installed for both contexts")
 				sort.Strings(contexts)
-				for i := 0; i < len(installedPlugins); i++ {
-					if i < numberOfPluginsToInstall {
-						Expect(installedPlugins[i].Context).Should(Equal(contexts[0]))
-					} else {
-						Expect(installedPlugins[i].Context).Should(Equal(contexts[1]))
-					}
-				}
+				Expect(f.ValidateInstalledPluginsOrder(contexts, installedPlugins)).To(BeTrue())
```

Comment:
What is the benefit of using `ValidateInstalledPluginsOrder` instead of normal validation?

Reply:
This test failed when both tmc and k8s context data was returned. So I had to refactor this to a method